t/023-rewrite/duplicate.t

```diff
@@ -0,0 +1,303 @@
+
+    location = /t {
+        echo "Hello /t";
+    }
+
+    location /t2 {
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.print("Hello t2") }
+    }
+--- request
+GET /t
+--- response_body
+Hello /t
+--- no_error_log
+rewrite 1 at location
+--- error_log
+rewrite 1 at server
+rewrite 2 at server
+
+
+
+=== TEST 10: rewrite directives at different scopes (server + location)
+--- config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at server") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at server") }
+
+    location = /t {
+        echo "Hello /t";
+    }
+
+    location /t2 {
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.print("Hello /t2\n") }
+    }
+--- request
+GET /t2
+--- response_body
+Hello /t2
+--- no_error_log
+rewrite 1 at server
+rewrite 2 at server
+--- error_log
+rewrite 1 at location
+
+
+
+=== TEST 11: rewrite directives at different scopes (http + location)
+--- http_config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at http") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at http") }
+--- config
+    location = /t {
+        echo "Hello /t";
+    }
+
+    location /t2 {
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.print("Hello /t2\n") }
+    }
+--- request
+GET /t
+--- response_body
+Hello /t
+--- no_error_log
+rewrite 1 at location
+--- error_log
+rewrite 1 at http
+rewrite 2 at http
+
+
+
+=== TEST 12: rewrite directives at different scopes (http + server)
+--- http_config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at http") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at http") }
+--- config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at server") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at server") }
+
+    location = /t {
+        echo "Hello /t";
+    }
+--- request
+GET /t
+--- response_body
+Hello /t
+--- no_error_log
+rewrite 1 at http
+rewrite 2 at http
+--- error_log
+rewrite 1 at server
+rewrite 2 at server
+
+
+
+=== TEST 13: rewrite directives at different scopes (http + server + location)
+--- http_config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at http") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at http") }
+--- config
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at server") }
+    rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at server") }
+
+    location = /t {
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 1 at location") }
+        rewrite_by_lua_block { ngx.log(ngx.ERR, "rewrite 2 at location") }
+        echo "Hello /t";
+    }
+--- request
+GET /t
+--- response_body
+Hello /t
+--- error_log
+rewrite 1 at location
+rewrite 2 at location
+--- no_error_log
+rewrite 1 at http
+rewrite 2 at http
+rewrite 1 at server
+rewrite 2 at server
```

Comment:
Need a test case reading request bodies via `ngx.req.read_body()` in the rewrite phase handler. It might be a slightly different kind of "yielding" :)